lib/components/gzip.js

```diff
@@ -15,8 +15,22 @@ Component.prototype.name = '__gzip__';
 
 Component.prototype.encode = function(key, msg) {
     let msgString = JSON.stringify(msg);
-    let usingGzip = Buffer.byteLength(msgString, 'utf8') >= GZIP_BYTE_THRESHOLD;
-    let convertedData = usingGzip ? zlib.gzipSync(msgString) : Buffer.from(msgString, 'utf-8');
+    let originalMsgSize = Buffer.byteLength(msgString, 'utf8');
+    let usingGzip = false;
+
+    let convertedData;
+    if (originalMsgSize >= GZIP_BYTE_THRESHOLD) {
+      let gzipData = zlib.gzipSync(msgString);
```

Comment:
The initial version of the software had a bug that processed all messages with a length greater than 125.
As a result, we processed numerous messages with 125 bytes or more, but the CPU impact was not noticeable.

To further improve the system, it would be beneficial to make the threshold dynamic. This means constantly adjusting its value to minimize the amount of unused gzip compression. By doing so, we can enhance the efficiency of the compression process and reduce unnecessary overhead.